pkg/v2/resource/chart/resource.go

```diff
@@ -101,6 +105,36 @@ func (r *Resource) Name() string {
 	return Name
 }
 
+func (r *Resource) getGuestHelmClient(ctx context.Context, obj interface{}) (helmclient.Interface, error) {
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for chart-operator chart in the guest cluster")
+
+	clusterConfig, err := prepareClusterConfig(r.baseClusterConfig, clusterGuestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	guestHelmClient, err := r.guest.NewHelmClient(ctx, clusterConfig.ClusterID, clusterConfig.Domain.API)
+	if guestcluster.IsNotFound(err) {
```

Comment:
Is this `getGuestHelmClient()` used in many places? Would it make sense to lift this `if guestcluster.IsNotFound(err)...` case to calling site so canceling of resource reconciliation is more apparent when reading resource's reconciliation logic? I don't have strong opinion on this, just something that came to my mind.

Reply:
Makes sense to me, this will be called in each of the resource current/create/update/delete methods, but I think enhancing visibility is worth the repetition in this case, PTAL.